apps/meteor/client/views/marketplace/AppsProvider.tsx

```diff
@@ -154,17 +156,37 @@ const AppsProvider: FC = ({ children }) => {
 		reload: async () => undefined,
 	});
 
-	const fetch = useCallback(async (): Promise<void> => {
+	const [privateAppsState, dispatchPrivateApps] = useReducer<
+		Reducer<
+			AsyncState<{ apps: App[] }> & {
+				reload: () => Promise<void>;
+			},
+			Action
+		>
+	>(reducer, {
+		phase: AsyncStatePhase.LOADING,
+		value: undefined,
+		error: undefined,
+		reload: async () => undefined,
+	});
+
+	const isAdminUser = usePermission('manage-apps');
+
+	const fetch = useCallback(async (isAdminUser?: string): Promise<void> => {
```

Comment:
great opportunity to use usequery

Reply:
I have a refactor of this component on the aims for a long time, even before we started using tanstack-query, the problem is that the comes and goes of sprints never allowed me to do it (especially because since last month I was the only one working on this).